opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java

```diff
@@ -67,29 +70,27 @@ public synchronized void processClient(List<EventClient> eventClientList) {
     }
 
     public void processClient(List<EventClient> eventClients, boolean localEvents) {
-
         ClientClassification clientClassification = assetJsonToJava("ec_client_classification.json", ClientClassification.class);
-
         if (clientClassification == null) {
             return;
         }
+
         Location operationalArea = Utils.getOperationalAreaLocation(PreferencesUtil.getInstance().getCurrentOperationalArea());
         String operationalAreaLocationId = operationalArea == null ? null : operationalArea.getId();
-        boolean hasSynchedEventsInTarget = false;
-
+        boolean hasSyncedEventsInTarget = false;
         if (!eventClients.isEmpty()) {
+            String operationalAreaId = null;
             for (EventClient eventClient : eventClients) {
                 Event event = eventClient.getEvent();
                 if (event == null || event.getEventType() == null) {
                     continue;
                 }
+
                 String eventType = event.getEventType();
                 if (eventType.equals(SPRAY_EVENT)) {
-                    String operationalAreaId = processSprayEvent(event, clientClassification, localEvents);
-                    if (!hasSynchedEventsInTarget && operationalAreaLocationId != null &&
```

Comment:
Return this code, or move it to a function so that its called for the `SPRAY_EVENT` and `MOSQUITO_COLLECTION_EVENT` Its setting a boolean flag, removing it introduces a bug

Reply:
Moving the conversation here https://github.com/OpenSRP/opensrp-client-reveal/pull/121/files/c8bf688e2dcdf461652e3bbc9952365e5ba2909d#r267655879.